Realm/Realm.Unity/Editor/Common/RealmWeaver.Schema.cs

```diff
@@ -119,6 +119,14 @@ private MethodDefinition GetModuleInitializer()
                 HasThis = false
             };
 
+#if UNITY_IOS || ENABLE_IL2CPP
+            var isUnityEngineAvailable = _moduleDefinition.AssemblyReferences.Any(a => a.Name.Contains("UnityEngine"));
+            if (isUnityEngineAvailable)
+            {
+                var attributeConstructor = _moduleDefinition.ImportReference(typeof(UnityEngine.RuntimeInitializeOnLoadMethodAttribute).GetConstructor(Type.EmptyTypes));
```

Comment:
We don't want to assume that the build pipeline has a dependency on UnityEngine. Instead, we should construct the type and the constructor reference by hand, similar to what we're doing for Fody.PropertyChanged:

https://github.com/realm/realm-dotnet/blob/4f900bdda9bc1ebbdb5f8ae9088895532e1c30fc/Realm/Realm.Unity/Editor/Common/ImportedReferences.cs#L372-L373